File: backend/app/schemas/resource_search.py
from __future__ import annotations

from pydantic import BaseModel, Field


class ResourceSearchSourceItemOut(BaseModel):
    key: str = Field(min_length=1, max_length=32)
    enabled: bool
    server: str | None = Field(default=None, max_length=512)
    username: str | None = Field(default=None, max_length=256)
    password: str | None = Field(default=None, max_length=256)
    token: str | None = Field(default=None, max_length=2048)


class ResourceSearchSourceListOut(BaseModel):
    sources: list[ResourceSearchSourceItemOut] = []


class ResourceSearchSourceUpdateIn(BaseModel):
    enabled: bool | None = None
    server: str | None = Field(default=None, max_length=512)
    username: str | None = Field(default=None, max_length=256)
    password: str | None = Field(default=None, max_length=256)
    token: str | None = Field(default=None, max_length=2048)


class TaskSuggestionItemOut(BaseModel):
    taskname: str = Field(default="", max_length=512)
    shareurl: str = Field(default="", max_length=2048)
    content: str | None = None
    datetime: str | None = None
    channel: str | None = None
    source: str | None = None
    # Episode grouping fields (added for series grouping)
    group_key: str | None = None
    episode: str | None = None
    season: str | None = None
    series_name: str | None = None


class TaskSuggestionListOut(BaseModel):
    success: bool = True
    data: list[TaskSuggestionItemOut] = []
    message: str | None = None

File: frontend/src/types/resourceSearch.ts
export type ResourceSearchSourceKey = 'net' | 'cloudsaver' | 'pansou'

export type ResourceSearchSourceItem = {
  key: ResourceSearchSourceKey
  enabled: boolean
  server?: string | null
  username?: string | null
  password?: string | null
  token?: string | null
}

export type ResourceSearchSourceListResponse = {
  sources: ResourceSearchSourceItem[]
}

export type TaskSuggestionItem = {
  taskname: string
  shareurl: string
  content?: string | null
  datetime?: string | null
  channel?: string | null
  source?: string | null
  verify?: boolean | null
  // Episode grouping fields (added for series grouping)
  group_key?: string | null
  episode?: string | null
  season?: string | null
  series_name?: string | null
}

export type TaskSuggestionResponse = {
  success: boolean
  data: TaskSuggestionItem[]
  message?: string | null
}

